src/main/java/io/nats/client/support/JwtUtils.java

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.client.support;
+
+import io.nats.client.NKey;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.MessageDigest;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Base64;
+
+import static io.nats.client.support.JsonUtils.beginJson;
+import static io.nats.client.support.JsonUtils.endJson;
+
+public class JwtUtils {
+    static class Nats implements JsonSerializable {
+        String issuerAccount;
+        String[] tags;
+
+        @Override
+        public String toJson() {
+            StringBuilder sb = beginJson();
+            JsonUtils.addField(sb, "issuer_account", issuerAccount);
+            JsonUtils.addStrings(sb, "tags", Arrays.asList(tags));
+            JsonUtils.addField(sb, "type", "user");
+            JsonUtils.addField(sb, "version", 2);
+            return endJson(sb).toString();
+        }
+    }
+
+    static class Claim implements JsonSerializable {
+        Duration exp;
+        long iat;
+        String iss;
+        String jti;
+        String name;
+        Nats nats;
+        String sub;
+
+        @Override
+        public String toJson() {
+            StringBuilder sb = beginJson();
+            if (exp != null && !exp.isZero() && !exp.isNegative()) {
+                long unixSeconds = exp.toMillis() / 1000;
+                JsonUtils.addField(sb, "exp", unixSeconds);
```

Comment:
This needs to be `iat + unixSeconds`... since the duration represents how long the token should be valid relative to the issued at time.

Reply:
@brimworks can you sort this out with @matthiashanel

Reply:
agreed on `iat + unixSeconds`
from the adr:
 // expiration (when specified) time in unix seconds, derived from provided duration.

Reply:
Sounds like we do need the `iat + unixSeconds`.

Reply:
done